internal/commands/manifest_create.go

```diff
@@ -1,99 +1,73 @@
-		Long: "manifest create generates a manifest list for a multi-arch image",
+		Use:   "create <manifest-list> <manifest> [<manifest> ... ] [flags]",
+		Args:  cobra.MatchAll(cobra.MinimumNArgs(2), cobra.OnlyValidArgs),
+		Short: "manifest create generates a manifest list for a multi-arch image",
+		Example: `pack manifest create cnbs/sample-package:hello-multiarch-universe \
+		cnbs/sample-package:hello-universe \
+		cnbs/sample-package:hello-universe-windows`,
+		Long: `Create a manifest list or image index for the image to support muti architecture for the image, it create a new ManifestList or ImageIndex with the given name and adds the list of Manifests to the newly created ImageIndex or ManifestList
+		
+		If the <manifest-list> already exists in the registry: pack will save a local copy of the remote manifest list,
+		If the <manifest-list> doestn't exist in a registry: pack will create a local representation of the manifest list that will only save on the remote registry if the user publish it`,
 		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
-			if err := validateManifestCreateFlags(&flags); err != nil {
-				return err
-			}
-
-			mediaType, err := validateMediaTypeFlag(flags.Format)
-			if err != nil {
-				return err
-			}
-
-			packHome, err := config.PackHome()
-			if err != nil {
-				return err
-			}
-
-			manifestDir := filepath.Join(packHome, "manifests")
-
-			indexName := args[0]
+			imageIndex := args[0]
 			manifests := args[1:]
-			if err := pack.CreateManifest(cmd.Context(), client.CreateManifestOptions{
-				ManifestName: indexName,
-				Manifests:    manifests,
-				MediaType:    mediaType,
-				Publish:      flags.Publish,
-				Registry:     flags.Registry,
-				ManifestDir:  manifestDir,
-			}); err != nil {
-				return err
-			}
-
-			logger.Infof("Successfully created image index %s", style.Symbol(indexName))
-
-			return nil
+			return pack.CreateManifest(
+				cmd.Context(),
+				imageIndex,
+				manifests,
+				client.CreateManifestOptions{
+					Format:   flags.format,
+					Registry: flags.registry,
+					Insecure: flags.insecure,
+					Publish:  flags.publish,
+					All:      flags.all,
+				},
+			)
 		}),
 	}
 
-	cmd.Flags().BoolVar(&flags.Publish, "publish", false, `Publish to registry`)
-	cmd.Flags().BoolVar(&flags.Insecure, "insecure", false, `Allow publishing to insecure registry`)
-	cmd.Flags().StringVarP(&flags.Format, "format", "f", "v2s2", `Format to save image index as ("OCI" or "V2S2")`)
-	cmd.Flags().StringVarP(&flags.Registry, "registry", "r", "", `Registry URL to publish the image index`)
+	cmdFlags := cmd.Flags()
 
-	AddHelpFlag(cmd, "create")
-	return cmd
-}
-
-func validateManifestCreateFlags(p *ManifestCreateFlags) error {
-	if p.Format == "" {
-		return errors.Errorf("--format flag received an empty value")
+	cmdFlags.StringVarP(&flags.format, "format", "f", "v2s2", "Format to save image index as ('OCI' or 'V2S2') (default 'v2s2')")
```

Comment:
Remove `(default 'v2s2')` because it is printed duplicated after actually setting "v2s2" as default
![Screenshot 2024-02-09 at 11 06 34 AM](https://github.com/husni-faiz/pack/assets/1181799/66e9c072-1965-4210-9174-a5efb80b1497)